src/screens/MoreScreen.js

```diff
@@ -1,33 +1,95 @@
+import {AppLoading} from 'expo';
 import React, {Component} from 'react';
-import {Text} from 'react-native';
-import {Icon} from 'react-native-elements';
+import {Platform, Text, View} from 'react-native';
+import {connect} from 'react-redux';
+import {Icon, Button, Card} from 'react-native-elements';
 
-import {CardSection, Card} from '../components/common/index';
+import {logUserOutOfFacebook} from '../actions';
+
+import {more_screen} from '../locale.en';
+import {navKeys} from '../constants';
 
 class MoreScreen extends Component {
+    componentDidMount() {
+        const {token, navigation: {navigate, state: {key}}} = this.props;
+
+        if (!token && key !== navKeys.LOGIN) {
+            navigate(navKeys.LOGIN);
+        }
+
+        this.props.navigation.setParams({onLogout: this.onLogout});
```

Comment:
Similar to the work we did on `SearchScreen` last week, we pass the `onLogout` method to the static navigation props below so our static component can trigger an action with a button press.